ui/app/templates/components/generate-credentials-totp.hbs

```diff
@@ -0,0 +1,51 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<PageHeader as |p|>
+  <p.top>
+    <Hds::Breadcrumb>
+      <Hds::Breadcrumb::Item
+        @text={{@backendPath}}
+        @route="vault.cluster.secrets.backend"
+        @model={{@backendPath}}
+        data-test-link="role-list"
+      />
+      <Hds::Breadcrumb::Item @text="Codes" @route="vault.cluster.secrets.backend" @model={{@backendPath}} />
+      <Hds::Breadcrumb::Item @text={{@keyName}} @route="vault.cluster.secrets.backend.show" @model={{@keyName}} />
+      <Hds::Breadcrumb::Item @text={{this.title}} @current={{true}} />
+    </Hds::Breadcrumb>
+  </p.top>
+  <p.levelLeft>
+    <h1 data-test-title class="title is-3">
+      {{this.title}}
+    </h1>
+  </p.levelLeft>
+</PageHeader>
+
+<div
+  class="box is-fullwidth is-sideless is-paddingless is-marginless"
+  {{did-insert this.startTimer}}
+  {{will-destroy this.cancelTimer}}
+>
+  <MessageError @model={{this.model}} />
+  <InfoTableRow @label="Code">
+    <MaskedInput @name="code" @value={{@model.code}} @displayOnly={{true}} @allowCopy={{true}} />
+    <label class="code-validity">
+      <div>Valid for {{this.remainingTime}} seconds</div>
+      <progress id="code-validity" value="{{this.elapsedTime}}" max="{{@model.period}}"></progress>
+    </label>
+  </InfoTableRow>
+</div>
+<div class="field is-grouped box is-fullwidth is-bottomless">
+  <div class="control">
+    <Hds::Button
+      @text="Back"
+      @color="secondary"
+      @route="vault.cluster.secrets.backend.list-root"
+      @model={{this.backendPath}}
+      data-test-secret-generate-back={{true}}
```

Comment:
nit -- we don't need =true on the tests selectors anymore (that is leftover from earlier versions of Ember when we did)
```suggestion
      data-test-secret-generate-back
```